content/src/content/docs/learn/tutorials/first-endpoint.mdx

```diff
@@ -0,0 +1,355 @@
+provided.
+
+We can `pipe` as many `Effect`s together as we like:
+
+```ts twoslash ins={3-5} showLineNumbers=false
+import { Effect } from "effect"
+// ---cut---
+const effect = Effect.sync(() => "Hello, world!").pipe(
+  Effect.map((s) => s.toUpperCase()),
+  Effect.map((s) => s.split("")),
+  Effect.map((s) => s.reverse()),
+  Effect.map((s) => s.join("")),
+)
+const result = await Effect.runSync(effect)
+console.log(result)
+// => "!DLROW ,OLLEH"
+```
+
+This should give us just enough to carry on with migrating our first endpoint
+to Effect. As we continue through this tutorial, we'll introduce more and more
+things you can do with the `Effect` type.
+
+## Understanding `HttpRouter`
+
+Looking back at our `HttpRouter`, we used `pipe` to add a new endpoint to our
+app:
+
+```ts twoslash showLineNumbers=false
+import { HttpRouter, HttpServerResponse } from "@effect/platform"
+import { Effect } from "effect"
+const health = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+// ---cut---
+const router = HttpRouter.empty.pipe(
+  HttpRouter.get("/health", health),
+)
+```
+
+`HttpRouter` is a data structure that represents a collection of routes. The
+simplest router is one with no routes at all, and Effect exposes that to us as
+the `HttpRouter.empty` value. Under the hood, this is itself an `Effect`:
+
+```ts twoslash showLineNumbers=false
+import { HttpRouter } from "@effect/platform"
+import { Effect } from "effect"
+// ---cut---
+console.log(Effect.isEffect(HttpRouter.empty))
+// => true
+```
+
+The helper `HttpRouter.get` takes an `HttpRouter` as an argument and returns a
+new `HttpRouter` with the given route added. If we wanted to, we could have
+called it like a regular function. The following two lines of code do exactly
+the same thing:
+
+```ts twoslash showLineNumbers=false
+import { HttpRouter, HttpServerResponse } from "@effect/platform"
+import { Effect } from "effect"
+const health = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+// ---cut---
+HttpRouter.empty.pipe(HttpRouter.get("/health", health))
+HttpRouter.get("/health", health)(HttpRouter.empty)
+```
+
+If we add a few more routes, we can see how this gets quickly out of hand:
+
+```ts twoslash showLineNumbers=false
+import { HttpRouter, HttpServerResponse } from "@effect/platform"
+import { Effect } from "effect"
+const health = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+const status = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+const version = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+// ---cut---
+HttpRouter.get("/health", health)(
+  HttpRouter.get("/status", status)(
+    HttpRouter.get("/version", version)(
+      HttpRouter.empty,
+    ),
+  ),
+)
+
+// vs
+
+HttpRouter.empty.pipe(
+  HttpRouter.get("/version", version),
+  HttpRouter.get("/status", status),
+  HttpRouter.get("/health", health),
+)
+```
```

Comment:
Suggest two snippets here. The first one could leads with "... see how this gets out of hand..." and the second leads with something like "versus using `pipe`":